text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,457 @@
+$ cdk init --git-url [URL] --template-name [name]
+
+# Examples:
+# Using GitHub URL
+$ cdk init --git-url https://github.com/username/my-cdk-templates.git --language=[csharp|fsharp|go|java|javascript|python|typescript]
+
+# Using BitBucket URL
+$ cdk init --git-url https://bitbucket.org/username/my-cdk-templates.git --language=[csharp|fsharp|go|java|javascript|python|typescript]
+```
+
+[This](#project-file-structure-after-running-cdk-init) is how your project’s file structure should look after running `cdk init` in a target directory called “my-cdk-project”.
+
+#### Using NPM Package Templates
+
+Create projects with templates pulled from NPM packages (on [npmjs.com](http://npmjs.com/) or any registry that hits NPM endpoint):
+
+```
+# Use specific template from NPM package with many templates
+$ cdk init --npm-package [package-name] --template-name [name] --language=[csharp|fsharp|go|java|javascript|python|typescript]
+
+# If NPM package only contains one template, don't need to specify template name
+$ cdk init --npm-package [package-name] --language=[csharp|fsharp|go|java|javascript|python|typescript]
+
+# Examples:
+# Using specific template from NPM package
+$ cdk init --npm-package my-cdk-templates --template-name my-template --language=typescript
+
+# Using NPM package with single template
+$ cdk init --npm-package my-cdk-template --language=typescript
+```
+
+[This](#project-file-structure-after-running-cdk-init) is how your project’s file structure should look after running `cdk init` in a target directory called “my-cdk-project”.
+
+##### Project File Structure After Running `cdk init` 
+
+```
+my-cdk-project/
+├── bin/       # Entry point for your CDK app
+│   └── mycdk-project         # language-specific file that instantiates new CDK app, creates your stack, adds it to your app, and gives the stack an ID
+├── lib/       # Stack definitions and constructs
+│   └── my-cdk-project-stack  # language-specific file that defines CDK stack class where you define your AWS resources (S3 Buckets, Lambda functions, etc))
+├── test/      # CDK Unit tests
+│   └── my-cdk-project.test   # language-specific file that verifies your infrastructure code generates the expected CloudFormation resources
+├── cdk.json   # CDK configuration file that specifies how to run your app
+├── README.md  # Project documentation
+└── # language-specific dependency management, language configuration, and testing configuration files
+```
+
+##### Troubleshooting
+
+If a custom template fails to load, the CLI will throw an error. Below are some basic troubleshooting steps for possible error scenarios associated with using custom templates:
+
+###### Template not found:
+
+    * Verify the path/URL/package name is correct and public
+
+###### Permission errors:
+
+    * Ensure you have write permissions in the target directory running `cdk init`
+    * Check that the template files have appropriate permissions
```

Comment:
True, but these two examples are highly unlikely to occur, so unlikely I wouldn't even mention them. I would however mention git/npm permission errors that might happen, and explain how credentials need to be setup.